src/ast/semantic_analyser.cpp

```diff
@@ -758,6 +761,19 @@ void SemanticAnalyser::visit(Binop &binop)
           }
         }
       }
+      else if (lval < 0 || rval < 0) {
+        // SDIV is not implemented for bpf. See Documentation/bpf/bpf_design_QA
+        // in kernel sources
+        switch (binop.op) {
+          case bpftrace::Parser::token::DIV:
+            buf << "signed division can lead to undefined behavior"
+                << std::endl;
```

Comment:
This will actually insert a double new line in the message as the printer already does it. I'll make a PR to remove trailing newline from the message passed into the printer.